pandas/core/indexes/datetimelike.py

```diff
@@ -651,6 +652,15 @@ def __add__(self, other):
                 raise TypeError("cannot add {typ1} and {typ2}"
                                 .format(typ1=type(self).__name__,
                                         typ2=type(other).__name__))
+            elif isinstance(other, np.ndarray):
+                if is_timedelta64_dtype(other):
+                    return self._add_delta(TimedeltaIndex(other))
+                elif is_integer_dtype(other):
+                    return NotImplemented
+                else:
+                    raise TypeError("cannot add {typ1} and np.ndarray[{typ2}]"
```

Comment:
Same question as with the `NotImplemented`.  Let's make sure we have a test to hit this branch.